File: bench/harmonic/harmonic.lua
-- The Great Computer Language Shootout
-- http://shootout.alioth.debian.org/
--
-- contributed by Isaac Gouy


local n = tonumber(arg and arg[1]) or 10000000

local d, partialSum = 0.0, 0.0
local i = 0

while i < n do
   d = d + 1.0
   partialSum = partialSum + (1.0/d)
   i = i + 1
end

io.write(string.format("%0.9f",partialSum), "\n")

File: bench/harmonic/harmonic.vb
' The Computer Language Shootout
' http://shootout.alioth.debian.org/
' contributed by Isaac Gouy

Imports System

Public Module Harmonic
   Sub Main(ByVal CmdArgs() As String)
      Dim N As Integer = Int32.Parse(CmdArgs(0)) 
      Dim PartialSum As Double = 0.0
      For I As Integer = 1 To N 
         PartialSum += 1.0/I
      Next
      Console.WriteLine("{0:f9}", PartialSum)
   End Sub
End Module
